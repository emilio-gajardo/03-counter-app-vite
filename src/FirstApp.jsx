import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subtitle}</h2>
      <h2>{subtitle}</h2>
      <h3>{name}</h3>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

FirstApp.defaultProps = {
  title: 'No title',
  subtitle: 'No subtitle',
  name: 'Joham',
}
