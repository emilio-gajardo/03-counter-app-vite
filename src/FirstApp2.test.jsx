import { render, screen, cleanup } from "@testing-library/react";
import { FirstApp } from "./FirstApp";

describe('Test: <FirstApp />', () => {

    afterEach(cleanup);// Limpiar recursos
    let title = 'Title - testing';
    let subtitle = 'Subtitle - testing';

    test('1) Debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test(`2) Debe mostrar el mensaje "${title}"`, () => {
        render(<FirstApp title={title} />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('3) Debe mostrar title en h1', () => {
        // expect(getByText(title)).toBeTruthy();
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('4) Debe mostrar el subtitle enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle} />);
        expect(screen.getAllByText(subtitle)).toBeTruthy();
    });


});