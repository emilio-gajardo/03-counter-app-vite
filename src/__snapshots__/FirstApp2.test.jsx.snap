// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: <FirstApp /> 1) Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Title - testing
  </h1>
  <h2>
    No subtitle
  </h2>
  <h2>
    No subtitle
  </h2>
  <h3>
    Joham
  </h3>
</div>
`;
